EIPS/eip-5553.md

```diff
@@ -0,0 +1,260 @@
+        recToken = _recAddress;
+        metadataURI = _params.metadataUri;
+        fileHash = _params.fileHash;
+        tokenId = _tokenId;
+        
+        _safeMint(_songLedger, _tokenId);
+        emit Minted(_params.shortName,_songLedger,_compAddress,_recAddress,_msgSender(),tokenId,_params.metadataUri);
+    }
+
+    function changeMetadataURI(string memory _newURI,string memory _newFileHash) public 
+     {
+        string memory oldURI = metadataURI;
+        string memory oldHash = fileHash;
+        metadataURI = _newURI; 
+        fileHash = _newFileHash;
+        
+        emit MetadataChanged(oldURI, oldHash,_newURI,_newFileHash);
+    }
+    
+    function royaltyInterestTokens() external view returns (address[] memory) {
+        address[] memory items = new address[](2); 
+        items[0] = compToken;
+        items[1] = recToken;
+        return items;
+    }
+    function ledger() external view returns (address) {
+         return songLedger;
+    }
+
+    event MetadataChanged(
+        string  oldUri, string oldFileHash,
+        string  newUri, string newFileHash
+        );
+    event Minted(
+        string  abbvName,
+        address ledger,
+        address compToken,
+        address recToken,
+        address creator,
+        uint256 tokenId,
+        string metadataUri
+        );
+}
+
+
+
+```
+
+#### Deploying a new music works instance using a simple song registry contract
+
+```solidity  
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "./SongRegistration.sol";
+import "./CompositionRoyaltyToken.sol";
+import "./RecordingRoyaltyToken.sol";
+
+
+contract SimpleSongLedger is IERC721Receiver {
+    using Counters for Counters.Counter;
+    Counters.Counter private songIds;
+      function onERC721Received(address, address, uint256, bytes calldata) external pure returns (bytes4) {
+        return IERC721Receiver.onERC721Received.selector;
+    }
+
+    function mintSong(SongMintingParams memory _params) public {
+        CompositionRoyaltyToken comp = new CompositionRoyaltyToken(address(this),"SONGCOMP","COMP");
+        RecordingRoyaltyToken rec = new RecordingRoyaltyToken(address(this),"SONGREC","REC");
+        songIds.increment();
+
+        SongRegistration newSong = new SongRegistration(
+                                        songIds.current(),
+                                        address(this),
+                                        _params,
+                                        address(comp),
+                                        address(rec)
+                                    );
+    }
+}
+
+
+```
+
+## Rationale
+
+### A more robust royalty interest structure
+
+Current royalty implementations deal only with a single type of royalty payments: NFT sales. They also only allow a single type of royalty - i.e you cannot have different royalty interest owners receive royalty based on different licensing types. 
```

Comment:
Rationale explains how design decisions are made, motivation explains why author proposes this EIP and why having such EIP is better than not having one. I feel this 's more aligned with motivation

Reply:
Fixed working and moved one subsection to motivation.